### ISSUE TITLE
Copy constructor should not be allowed to omit this initializer call in a type that is not a record

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#68345]
According to https://github.com/dotnet/csharplang/blob/main/proposals/primary-constructors.md#primary-constructors-on-records:

> If an explicit constructor declaration is a "copy constructor" - a constructor that takes a single parameter of the enclosing type - it is not required to call a this initializer, and will not execute the member initializers present in the record declaration.

A copy-constructor in a *record* is allowed to not call `this` initializer. However, the compiler also allows it for non-record type with a primary constructor:

```C#
    public class D(int X)
    {
        protected D(D other)
        {
            Console.Write("copy ctor");
        }

        protected D(string other) // error CS8862: A constructor declared in a type with parameter list must have 'this' constructor initializer.
        {
            Console.Write("not a copy ctor");
        }
    }
```

CS8862 should be reported for the copy constructor as well in this case.
